### ISSUE TITLE
NET-1663 Fix S1172: code fix does not delete unused parameter in function call

### ISSUE THREAD

**Issue by urmel9** [SonarSource/sonar-dotnet#8187]
- Operating system: Microsoft Visual Studio Community 2022 (64-Bit) - Current Version 17.7.0
- SonarLint plugin version: 9.8.0.76515
- Programming language you're coding in: C#
- Is connected mode used: No

### Problem:
The Autofix of [S1172](https://rules.sonarsource.com/csharp/RSPEC-1172/) leads to a follw-up error. The rule and the description are correct, but the autofix function is wrong. It deletes the param in the private function but ignores the function calls and leads to an IDE error. 

### Solution:
Delete unused param also in function call
OR disable autofix for this rule.

Can you please fix this? Thank you :)

PS: Sadly, I could not reproduce the issue with the test code of the rule but literally two people in our team saw this problem :D

**Comment by sebastien-marichal** [comment]
Hello @urmel9,

Thank you for reaching out!

I tried to reproduce the issue based on your description, this is what I found/understood:

Considering this code where S1172 is raised:

```cs
DoSomething(21, 42);

void DoSomething(int a, int b) // S1172 - b is unsed
{
    Console.WriteLine(a);
}
```

As of today, when applying the code fix this is the result:

```cs
DoSomething(21, 42);
           // ^^^^ <- This is expected to be removed as well.

void DoSomething(int a)
{
    Console.WriteLine(a);
}
```

You are expecting the `DoSomething(21, 42);` call to become `DoSomething(21);` but is not updated, is that correct?

**Comment by urmel9** [comment]
Yea, exactly @sebastien-marichal ! :) 
Because otherwise the compiler throws an error because the function DoSomething is not overload (and expect 1 param). So you have to delete it by yourself

Nevertheless, this is a rule where we would turn off the autofix-feature because sometimes we just comment out certain variables :D But I raised another issue for that in the sonar community: https://community.sonarsource.com/t/new-feature-disable-autofix-for-specific-rules-in-editorconfig/102315/6

**Comment by sebastien-marichal** [comment]
Thank you @urmel9 for the quick response!

I will add this issue to our backlog.

To fix the issue, [SymbolFinder.FindCallersAsync](https://learn.microsoft.com/en-us/dotnet/api/microsoft.codeanalysis.findsymbols.symbolfinder.findcallersasync?view=roslyn-dotnet-4.6.0#microsoft-codeanalysis-findsymbols-symbolfinder-findcallersasync(microsoft-codeanalysis-isymbol-microsoft-codeanalysis-solution-system-threading-cancellationtoken)) could be used.

**Comment by pavel-mikula-sonarsource** [comment]
Moved to [NET-1663](https://sonarsource.atlassian.net/browse/NET-1663)

[NET-1663]: https://sonarsource.atlassian.net/browse/NET-1663?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
